scripts/perftests/RunPerformanceTests.ps1

```diff
@@ -0,0 +1,178 @@
+    # The logs location is optional
+    function RunRestore([string]$solutionFilePath, [string]$nugetClient, [string]$resultsFile, [string]$logsPath, [string]$runName,
+            [switch]$cleanGlobalPackagesFolder, [switch]$cleanHttpCache, [switch]$cleanPluginsCache, [switch]$killMsBuildAndDotnetExeProcesses, [switch]$force)
+    {
+        Log "Running $nugetClient restore $nugetClient with cleanGlobalPackagesFolder:$cleanGlobalPackagesFolder cleanHttpCache:$cleanHttpCache cleanPluginsCache:$cleanPluginsCache killMsBuildAndDotnetExeProcesses:$killMsBuildAndDotnetExeProcesses force:$force"
+
+        # Do the required cleanup if necesarry
+        if($cleanGlobalPackagesFolder -Or $cleanHttpCache -Or $cleanPluginsCache)
+        {
+            if($cleanGlobalPackagesFolder -And $cleanHttpCache -And $cleanPluginsCache)
+            {
+                $localsArguments = "all"
+            }
+            elseif($cleanGlobalPackagesFolder -And $cleanHttpCache)
+            {
+                $localsArguments =  "http-cache global-packages"
+            }
+            elseif($cleanGlobalPackagesFolder)
+            {
+                $localsArguments =  "global-packages"
+            }
+            elseif($cleanHttpCache)
+            {
+                $localsArguments = "http-cache"
+            } 
+            else 
+            {
+                Log "Too risky to invoke a locals clear with the specified parameters." "yellow"
+            }
+
+            if($(IsClientDotnetExe $nugetClient))
+            {
+                . $nugetClient nuget locals -c $localsArguments *>>$null
+            }
+            else 
+            {
+                . $nugetClient locals -clear $localsArguments -Verbosity quiet
+            }
+        }
+
+        if($killMsBuildAndDotnetExeProcesses)
+        {
+            Stop-Process -name msbuild*,dotnet* -Force
+        }
+
+        $start=Get-Date
+        if($(IsClientDotnetExe $nugetClient))
+        {
+            $logs = . $nugetClient restore $solutionFilePath $forceArg
+        }
+        else 
+        {
+            $logs = . $nugetClient restore $solutionFilePath -noninteractive $forceArg
+        }
+        Log $logs
+        $end=Get-Date
+        $totalTime=$end-$start
+
+        if(!$logsPath)
+        {
+            $logFile = [System.IO.Path]::Combine($logsPath, "restoreLog-$([System.IO.Path]::GetFileNameWithoutExtension($solutionFilePath))-$(get-date -f yyyyMMddTHHmmssffff).txt")
+            OutFileWithCreateFolders $logFile $logs
+        }
+
+        $globalPackagesFolder = $Env:NUGET_PACKAGES
+        if(Test-Path $globalPackagesFolder)
+        {
+            $gpfNupkgFiles = GetAllPackagesInGlobalPackagesFolder $globalPackagesFolder
+            $gpfNupkgsSize = (($gpfNupkgFiles | Measure-Object -property length -sum).Sum/1000000)
+            $gpfFiles = GetFiles $globalPackagesFolder
+            $gpfFilesSize = (($gpfFiles | Measure-Object -property length -sum).Sum/1000000)
```

Comment:
nit: consider more readable names (even if they are more verbose) instead of `gpf` prefix.